// angular-authentication-service/dist/app/app.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var router_1 = require("@angular/router");
var authentication_service_1 = require("./services/authentication.service");
var globalvars_service_1 = require("./services/globalvars.service");
var angular_2_local_storage_1 = require("angular-2-local-storage");
var platform_browser_1 = require("@angular/platform-browser");
var material_1 = require("@angular/material");
// Observable class extensions
require("rxjs/add/observable/of");
// Observable operators
require("rxjs/add/operator/catch");
require("rxjs/add/operator/debounceTime");
require("rxjs/add/operator/distinctUntilChanged");
var AppComponent = (function () {
    function AppComponent(router, globalservice, localStorage, authservice, iconRegistry, sanitizer) {
        this.router = router;
        this.globalservice = globalservice;
        this.localStorage = localStorage;
        this.authservice = authservice;
        this.title = 'angular-authentication-service';
        this.position = 'below';
        this.authicon = 'login';
        iconRegistry.addSvgIcon('login', sanitizer.bypassSecurityTrustResourceUrl('assets/images/login.svg'));
        iconRegistry.addSvgIcon('logoff', sanitizer.bypassSecurityTrustResourceUrl('assets/images/logout.svg'));
    }
    AppComponent.prototype.ngOnInit = function () {
        if (localStorage.getItem('token') !== undefined && localStorage.getItem('token') !== null && localStorage.getItem('token') !== '') {
            this.authenticated = true;
        }
        else {
            this.authenticated = false;
        }
        this.authicon = (this.globalservice.isAuthenticated !== true) ? "lock_open" : "lock";
    };
    AppComponent.prototype.ngOnDestroy = function () {
        // prevent memory leak when component destroyed
        localStorage.setItem('token', '');
    };
    AppComponent.prototype.login = function (path) {
        if (path === void 0) { path = '/login'; }
        this.globalservice.isAuthenticated = true;
        this.authicon = "lock";
        console.log(path);
        this.router.navigate([path]);
    };
    AppComponent.prototype.logoff = function () {
        this.globalservice.isAuthenticated = true;
        this.authicon = "lock_open";
        console.log("logoff");
        this.router.navigate(['/login']);
    };
    AppComponent.prototype.isAuthenticated = function () {
        return this.authenticated;
    };
    AppComponent.prototype.print = function () {
    };
    return AppComponent;
}());
AppComponent = __decorate([
    core_1.Component({
        selector: 'app-root'
    }),
    __metadata("design:paramtypes", [router_1.Router,
        globalvars_service_1.GlobalVarsService,
        angular_2_local_storage_1.LocalStorageService,
        authentication_service_1.AuthenticationService,
        material_1.MdIconRegistry,
        platform_browser_1.DomSanitizer])
], AppComponent);
exports.AppComponent = AppComponent;
