// angular-authentication-service/dist/app/app.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var platform_browser_1 = require("@angular/platform-browser");
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var angular_2_local_storage_1 = require("angular-2-local-storage");
var flex_layout_1 = require("@angular/flex-layout");
var material_module_1 = require("./material.module");
require("hammerjs");
var authentication_service_1 = require("./services/authentication.service");
var globalvars_service_1 = require("./services/globalvars.service");
var auth_intercept_service_1 = require("./services/auth.intercept.service");
var login_component_1 = require("./login/login.component");
var login_1 = require("./classes/login");
var user_1 = require("./classes/user");
var http_intercept_1 = require("./classes/http.intercept");
var http_1 = require("@angular/common/http");
var animations_1 = require("@angular/platform-browser/animations");
var app_component_1 = require("./app.component");
var AuthenticationServiceModule = (function () {
    function AuthenticationServiceModule() {
    }
    return AuthenticationServiceModule;
}());
AuthenticationServiceModule = __decorate([
    core_1.NgModule({
        declarations: [
            app_component_1.AppComponent,
            login_component_1.LoginComponent
        ],
        imports: [
            platform_browser_1.BrowserModule,
            flex_layout_1.FlexLayoutModule,
            http_1.HttpClientModule,
            forms_1.FormsModule,
            forms_1.ReactiveFormsModule,
            animations_1.BrowserAnimationsModule,
            material_module_1.MaterialDesignModule,
            angular_2_local_storage_1.LocalStorageModule.withConfig({
                prefix: 'angular-authentication-service',
                storageType: 'localStorage'
            })
        ],
        exports: [
            material_module_1.MaterialDesignModule
        ],
        providers: [
            login_1.LogIn,
            user_1.User,
            globalvars_service_1.GlobalVarsService,
            auth_intercept_service_1.AuthIntercept,
            authentication_service_1.AuthenticationService,
            {
                provide: http_1.HTTP_INTERCEPTORS,
                useClass: http_intercept_1.HttpIntercept,
                multi: true,
            }
        ],
        bootstrap: [app_component_1.AppComponent]
    })
], AuthenticationServiceModule);
exports.AuthenticationServiceModule = AuthenticationServiceModule;
