// angular-authentication-service/dist/app/services/globalvars.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var login_1 = require("../classes/login");
var user_1 = require("../classes/user");
// Observable class extensions
require("rxjs/add/observable/of");
// Observable operators
require("rxjs/add/operator/catch");
require("rxjs/add/operator/debounceTime");
require("rxjs/add/operator/distinctUntilChanged");
require("rxjs/add/operator/toPromise");
var GlobalVarsService = (function () {
    function GlobalVarsService(LoginData, UserData) {
        this.LoginData = LoginData;
        this.UserData = UserData;
        this.AppName = 'MyApp';
        this.LoginData = new login_1.LogIn();
        this.UserData = new user_1.User();
        this.LoginData.username = '';
        this.LoginData.password = '';
        this.LoginData.app = this.AppName;
        this.UserData.access_token = '';
        this.UserData.firstName = '';
        this.UserData.lastName = '';
        this.UserData.groups = [];
        this.UserData.username = '';
    }
    GlobalVarsService.prototype.authenticated = function (isvalid) {
        this.isAuthenticated = isvalid;
    };
    return GlobalVarsService;
}());
GlobalVarsService = __decorate([
    core_1.Injectable(),
    __metadata("design:paramtypes", [login_1.LogIn, user_1.User])
], GlobalVarsService);
exports.GlobalVarsService = GlobalVarsService;
